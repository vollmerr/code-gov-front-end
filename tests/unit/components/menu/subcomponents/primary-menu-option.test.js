import React from 'react'
import { shallow } from 'enzyme'

import CustomLink from 'components/custom-link'
import PrimaryMenuOption from 'components/menu/subcomponents/primary-menu-option'

const props = {
  menuOption: { name: 'option-name', url: 'test-url' },
  onClick: jest.fn(),
}

let wrapper
describe('components - Menu - PrimaryMenuOption', () => {
  beforeEach(() => {
    wrapper = shallow(<PrimaryMenuOption {...props} />)
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should pass the `menuOption` to the `onClick`', () => {
    const event = {}
    wrapper.simulate('click', event)
    expect(props.onClick).toBeCalledWith(props.menuOption, event)
  })

  it('should render a `CustomLink` if passed a url that starts with `/`', () => {
    wrapper.setProps({ menuOption: { ...props.menuOption, url: '/test-url' } })
    expect(wrapper.find(CustomLink).length).toBe(1)
  })
})