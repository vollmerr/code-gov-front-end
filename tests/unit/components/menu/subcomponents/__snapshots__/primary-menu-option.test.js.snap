// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - Menu - PrimaryMenuOption should render correctly 1`] = `
<a
  aria-haspopup="true"
  className="nav-buttons"
  href="test-url"
  onClick={[Function]}
  role="menuitem"
  tabIndex="0"
>
  option-name
</a>
`;
